client/my-sites/checkout/src/test/checkout-main.tsx

```diff
@@ -669,4 +670,11 @@ describe( 'CheckoutMain', () => {
 		} );
 		expect( screen.getByText( 'Loading checkout' ) ).toBeInTheDocument();
 	} );
+
+	it( 'renders the promo code field', async () => {
+		render( <MockCheckout initialCart={ getPromoCart() } setCart={ mockSetCartEndpoint } /> );
+		await waitFor( () => {
+			expect( screen.getByText( /First-year promotional discount/ ) ).toBeInTheDocument();
```

Comment:
It might be better to use [.toBeVisible](https://github.com/testing-library/jest-dom#tobevisible) here since it ensures the following conditions are met:
```
it is present in the document
it does not have its css property display set to none
it does not have its css property visibility set to either hidden or collapse
it does not have its css property opacity set to 0
its parent element is also visible (and so on up to the top of the DOM tree)
it does not have the [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute
if <details /> it has the open attribute
```